api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -71,4 +79,12 @@ Table createTable(
    * @param to expected new name of the table.
    */
   void renameTable(TableIdentifier from, TableIdentifier to);
+
+  /**
+   * gets the table or throws {@link NoSuchTableException}
+   *
+   * @param tableIdentifier an identifier to identify this table in a namespace.
+   * @return instance of {@link Table} implementation referred by {@code tableIdentifier}
+   */
+  Table getTable(TableIdentifier tableIdentifier);
```

Comment:
Should be `loadTable` because `get` usually implies that the operation is a quick fetch, like a getter would be on a Java object.